src/Admin/ReaderThemes.php

```diff
@@ -210,17 +227,38 @@ public function get_default_reader_themes() {
 				]
 			);
 
-			if ( ! is_wp_error( $response ) ) {
-				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
 			}
-		}
 
-		if ( is_wp_error( $response ) ) {
-			return [];
-		}
+			/**
+			 * The response must minimally be an object with a themes array.
+			 *
+			 * @see https://wordpress.org/support/topic/issue-during-activating-the-updated-plugins/#post-13383737
+			 */
+			if (
+				is_wp_error( $response )
+				|| ! is_object( $response )
+				|| ! property_exists( $response, 'themes' )
+				|| ! is_array( $response->themes )
```

Comment:
In terms of performance and of logical progression, I think it makes more sense to start with the language construct `is_object()`. `is_object()` covers bigger ground than `is_wp_error()` (it entails that that is not true) and is faster.
```suggestion
				! is_object( $response )
				|| is_wp_error( $response )
				|| ! property_exists( $response, 'themes' )
				|| ! is_array( $response->themes )
```

Reply:
Actually, I think it makes sense to cover the full happy path first, and only check for the `WP_Error` last:
```suggestion
				! is_object( $response )
				|| ! property_exists( $response, 'themes' )
				|| ! is_array( $response->themes )
				|| is_wp_error( $response )
```
All three checks from the happy path are language constructs, whereas the `is_wp_error()` is a userland function (and thus slower), and it represents the "exceptional" case.